packages/table/src/Controls/TableSorter.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Button,
+  ButtonDropdown,
+  DropdownItem,
+  DropdownMenu,
+  DropdownToggle,
+  InputGroup,
+  Label,
+  UncontrolledTooltip,
+} from 'reactstrap';
+import Icon from '@availity/icon';
+import find from 'lodash/find';
+import sortBy from 'lodash/sortBy';
+import { TableSort } from '../types/TableSort';
+import { useTableContext } from '../TableContext';
+import { CurrentTableState, IdType, TableInstance } from '../types/ReactTable';
+import { TableSortOption } from '../types/TableSortOption';
+
+type Props = {
+  id?: string;
+  disabled?: boolean;
+  color?: string;
+  onSort?: (sortBy: TableSort[]) => void;
+  sortOptions?: TableSortOption[];
+  autoGenerateSortOptions?: boolean;
+} & React.HTMLAttributes<HTMLElement>;
+
+const TableSorter = <T extends IdType>({
+  id,
+  onSort,
+  disabled,
+  color,
+  sortOptions,
+  autoGenerateSortOptions,
+  ...rest
+}: Props): JSX.Element | null => {
+  const { sortableColumns, instance } = useTableContext();
+  const { toggleSortBy, state } = instance as TableInstance<T>;
+
+  const [isSortingDropdownOpen, setIsSortingDropdownOpen] = useState<boolean>(false);
```

Comment:
We can removeÏ each of these `useState`s that have `<boolean>`. The others should stay since they are initialized to `undefined`

`<TableSortOption | undefined>` can become `<TableSortOption>`. The `undefined` will be inferred from the initial value